velox/docs/functions/spark/array.rst

```diff
@@ -51,6 +51,13 @@ Array Functions
         SELECT array_min(ARRAY [4.0, float('nan')]); -- 4.0
         SELECT array_min(ARRAY [NULL, float('nan')]); -- NaN
 
+.. function:: array_remove(x, element) -> array
+
+    Remove all elements that equal ``element`` from array ``x``.
+
+        SELECT array_remove(ARRAY [1, 2, 3], 3); -- [1, 2]
+        SELECT array_remove(ARRAY [2, 1, NULL], 1); -- [2, NULL]
```

Comment:
Could we clarify the null behavior in the documentation? For example, what is the expected behavior of below case, and is Spark aligned with Presto?
> array_remove(ARRAY [2, 1, NULL], NULL)

Reply:
Sorry for late reply. And Spark result is same with presto.  I add the null behavior in the doc.
Spark:
```plain
spark-sql (default)> select array_remove(array(1,2,null),null);
NULL
Time taken: 2.762 seconds, Fetched 1 row(s)
```
Presto:
```plain
presto> select array_remove(array[1,2,3, null],null);
 _col0 
-------
 NULL  
(1 row)

Query 20240325_025644_00002_um9ey, FINISHED, 1 node
Splits: 17 total, 17 done (100.00%)
```

Reply:
Does any null input give null output, and what is the expected behavior of empty array, and an array whose elements are all null? Maybe better to clarify that in the documentation.
> Returns NULL if ...

Reply:
Yes, null output comes with null input.
* spark result:  
```plain
spark-sql (default)> select array_remove(array(), null);
NULL
Time taken: 3.058 seconds, Fetched 1 row(s)
spark-sql (default)> select array_remove(array(null, null), null);
NULL
Time taken: 0.137 seconds, Fetched 1 row(s)
spark-sql (default)> select array_remove(array(), 1);
[]
Time taken: 0.099 seconds, Fetched 1 row(s)
```
* presto result:
```plain
presto> select array_remove(array[], null);
 _col0 
-------
 NULL  
(1 row)

Query 20240326_142258_00001_e6ytf, FINISHED, 1 node
Splits: 17 total, 17 done (100.00%)
[Latency: client-side: 0:01, server-side: 0:01] [0 rows, 0B] [0 rows/s, 0B/s]

presto> select array_remove(array[null, null], null);
 _col0 
-------
 NULL  
(1 row)

Query 20240326_142317_00002_e6ytf, FINISHED, 1 node
Splits: 17 total, 17 done (100.00%)
[Latency: client-side: 263ms, server-side: 194ms] [0 rows, 0B] [0 rows/s, 0B/s]

presto> select array_remove(array[], 1);
 _col0 
-------
 []    
(1 row)

Query 20240326_142327_00003_e6ytf, FINISHED, 1 node
Splits: 17 total, 17 done (100.00%)
[Latency: client-side: 198ms, server-side: 178ms] [0 rows, 0B] [0 rows/s, 0B/s]
```

Reply:
@rui-mo thanks for your review. I have add some cases in the document:  
* element is null
* array with empty elements
* array with null elements

Reply:
Thanks.